### DISCUSSION TITLE
Rendering directives directly in a Jupyter notebook?

### DISCUSSION THREAD

**Discussion by stratisMarkou** [executablebooks/meta#1028] [Q&A]
The capability of Jupyter Books to render theorems, lemmas and other similar content, is awesome! Is there a way to have theorems written with directives be rendered directly in a notebook?

__In detail:__ Specifically, I would like to write a theorem (or lemma, proposition, proof, algorithm etc) with a sphinx directive in a Jupyter notebook, and have it rendered when I execute the markdown cell (without building the book). That way someone who downloads the notebook locally and runs the cell will see a human-readable rendered version rather than a bunch of latex wrapped in a directive. Is there any way to achieve this?

**Comment by agoose77** [comment]
You're probably looking for [jupyterlab-myst](https://github.com/executablebooks/jupyterlab-myst)! We're in the process of upgrading the extension for JupyterLab 4, but we have support for 3.x. I need to check if we've made a release with the theorem support for 3.x, I suspect not.

**Reply by rowanc1** [reply]
Jupyter Lab 4 is now supported, along with proofs!

![jupyterlab-proofs](https://github.com/executablebooks/meta/assets/913249/c4cd85be-6114-41f2-a6a5-a790a59e2b47)

**Reply by stratisMarkou** [reply]
That's _awesome_! Thanks for letting me know :)
